slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -344,73 +349,151 @@ trait MySQLProfile extends JdbcProfile { profile =>
       }
     }
 
-    override val stringJdbcType = new StringJdbcType {
+    class StringJdbcType extends super.StringJdbcType {
       override def valueToSQLLiteral(value: String) : String = {
         stringToMySqlString(value)
       }
     }
 
     import java.util.UUID
-
-    override val uuidJdbcType = new UUIDJdbcType {
-      override def sqlType = java.sql.Types.BINARY
+    class UUIDJdbcType extends super.UUIDJdbcType {
+      override def sqlType: Int = java.sql.Types.BINARY
       override def sqlTypeName(sym: Option[FieldSymbol]) = "BINARY(16)"
 
       override def valueToSQLLiteral(value: UUID): String =
         "x'"+value.toString.replace("-", "")+"'"
     }
 
-    override val instantType : InstantJdbcType = new InstantJdbcType {
+    trait MySQLTimeJdbcType [T] {
+
+      val min : T
+      val max : T
+      val serializeFiniteTime : (T => String)
+      val parseFiniteTime : (String => T)
+
+      @inline
+      private[this] val negativeInfinite = "-infinity"
+      @inline
+      private[this] val positiveInfinite = "infinity"
+
+      protected def serializeTime(time : T): String = {
+        time match {
+          case null => null
+          case `min` => negativeInfinite
+          case `max` => positiveInfinite
+          case _ => serializeFiniteTime(time)
+        }
+      }
+      protected def parseTime(time : String): T = {
+        time match {
+          case null => null.asInstanceOf[T]
+          case `negativeInfinite` => max
+          case `positiveInfinite` => min
+          case _ => parseFiniteTime(time)
+        }
+      }
+    }
```

Comment:
These methods are unused. Have you forget to use them instead of the default ones (from `case _ =>` clause)?

Reply:
You're right!
I have used the wrong method, fixed.